chia/server/node_discovery.py

```diff
@@ -251,14 +278,17 @@ async def _connect_to_peers(self, random) -> None:
                 while not got_peer and not self.is_closed:
                     sleep_interval = 1 + len(groups) * 0.5
                     sleep_interval = min(sleep_interval, self.peer_connect_interval)
+                    # Special case: try to find our first peer much quicker.
+                    if len(groups) == 0:
+                        sleep_interval = 0.1
```

Comment:
we won't try to connect the same peer more than once every 2 hours, is that correct? I think `sleep_interval` could be 0.1 seconds unconditionally, but I also think that's an unrelated change and maybe best reviewed in a separate PR.

(it would simplify some of this logic, because we could just make the default configuration for `peer_connect_interval` be 0.1 and just always use that)

Reply:
we need to at least profile that and see if it doesn't interfere with other cpu-intensive functions (for example sync). on the other side, if you have 0 peers, it's safe to keep the sleep small, as there's not too much else going on.